dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/mappers/OrgUnitDecorator.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.preheat.mappers;
+
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+
+public abstract class OrgUnitDecorator implements OrganisationUnitMapper
+{
+
+    private final OrganisationUnitMapper delegate;
+
+    protected OrgUnitDecorator( OrganisationUnitMapper delegate )
+    {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public OrganisationUnit map( OrganisationUnit organisationUnit )
+    {
+        if ( organisationUnit == null )
+        {
+            return null;
+        }
+
+        OrganisationUnit organisationUnit1 = delegate.map( organisationUnit );
+        organisationUnit1.setParent( fetchParent( organisationUnit.getParent() ) );
+        return organisationUnit1;
+    }
+
+    private OrganisationUnit fetchParent( OrganisationUnit organisationUnit )
+    {
+        if ( organisationUnit == null )
+        {
+            return null;
+        }
+
+        OrganisationUnit organisationUnit1 = new OrganisationUnit();
+        organisationUnit1.setUid( organisationUnit.getUid() );
```

Comment:
From the previous annotations I assume that this is what we have done before. Just wondering how would that affect different idSchemes? 🤔

Reply:
This is used to create the hierarchy of org units, I don't think this is using different idSchemes
But I am really not sure, maybe we should have a deeper look at it